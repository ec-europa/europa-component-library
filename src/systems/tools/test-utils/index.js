const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
