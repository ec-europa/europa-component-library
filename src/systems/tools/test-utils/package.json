{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
