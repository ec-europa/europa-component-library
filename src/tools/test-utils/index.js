const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
