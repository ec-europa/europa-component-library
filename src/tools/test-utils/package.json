{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
