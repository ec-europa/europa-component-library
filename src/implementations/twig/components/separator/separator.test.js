import { renderTwigFileAsNode, renderTwigFileAsHtml } from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'vitest-axe';

expect.extend(toHaveNoViolations);

describe('Separator', () => {
  const template = '@ecl/separator/separator.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render()).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = {
        extra_classes: 'custom-class custom-class--test',
      };

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      };

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, {}, true)),
      ).toHaveNoViolations();
    });
  });
});
