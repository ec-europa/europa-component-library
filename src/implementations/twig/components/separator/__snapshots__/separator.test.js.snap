// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > Default > renders correctly 1`] = `
<jest>
  <hr
    class="ecl-separator"
  />
</jest>
`;

exports[`Separator > Default > renders correctly with extra attributes 1`] = `
<jest>
  <hr
    class="ecl-separator"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  />
</jest>
`;

exports[`Separator > Default > renders correctly with extra class names 1`] = `
<jest>
  <hr
    class="ecl-separator custom-class custom-class--test"
  />
</jest>
`;
